catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/MetacardIngestNetworkPlugin.java

```diff
@@ -0,0 +1,174 @@
+            MetacardServices metacardServices, AttributeFactory attributeFactory,
+            MetacardCondition metacardCondition) {
+        this.keyValueParser = keyValueParser;
+        this.metacardServices = metacardServices;
+        this.attributeFactory = attributeFactory;
+        this.metacardCondition = metacardCondition;
+    }
+
+    public String getCriteriaKey() {
+        return metacardCondition.getCriteriaKey();
+    }
+
+    public void setCriteriaKey(String criteriaKey) {
+        metacardCondition.setCriteriaKey(criteriaKey);
+    }
+
+    public String getExpectedValue() {
+        return metacardCondition.getExpectedValue();
+    }
+
+    public void setExpectedValue(String expectedValue) {
+        metacardCondition.setExpectedValue(expectedValue);
+    }
+
+    public List<String> getNewAttributes() {
+        return newAttributes;
+    }
+
+    public void setNewAttributes(List<String> newAttributes) {
+        this.newAttributes = newAttributes;
+    }
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+        apply(input.getMetacards(), getClientInfoProperties());
+        return input;
+    }
+
+    //region - Passthrough  methods
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest input,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteRequest processPreDelete(DeleteRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse processPostDelete(DeleteResponse input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public QueryRequest processPreQuery(QueryRequest input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public QueryResponse processPostQuery(QueryResponse input) throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public ResourceRequest processPreResource(ResourceRequest input)
+            throws StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public ResourceResponse processPostResource(ResourceResponse input, Metacard metacard)
+            throws StopProcessingException {
+        return input;
+    }
+
+    //endregion
+
+    private void apply(List<Metacard> metacards, Map<String, Serializable> criteria) {
+        if (metacardCondition.applies(criteria)) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Metacard condition was true; expected {} of {} equals {}",
+                        metacardCondition.getCriteriaKey(),
+                        metacardCondition.getExpectedValue(),
+                        criteria.get(metacardCondition.getCriteriaKey()));
+            }
+            Map<String, String> parsedAttributes = keyValueParser.parsePairsToMap(newAttributes);
```

Comment:
I was trying to avoid having "redundant" class state but yes, this makes sense. It will look a tad strange because we still need the original `List` to support the metatype.